x/staking/keeper/grpc_query.go

```diff
@@ -229,12 +226,15 @@ func (k Querier) Delegation(ctx context.Context, req *types.QueryDelegationReque
 		return nil, err
 	}
 
-	delegation, err := k.GetDelegation(ctx, delAddr, valAddr)
+	delegation, err := k.Delegations.Get(ctx, collections.Join(sdk.AccAddress(delAddr), sdk.ValAddress(valAddr)))
 	if err != nil {
-		return nil, status.Errorf(
-			codes.NotFound,
-			"delegation with delegator %s not found for validator %s",
-			req.DelegatorAddr, req.ValidatorAddr)
+		if errors.Is(err, collections.ErrNotFound) {
+			return nil, status.Errorf(
+				codes.NotFound,
+				"delegation with delegator %s not found for validator %s",
+				req.DelegatorAddr, req.ValidatorAddr)
+		}
+		return nil, err
```

Comment:
Shouldn't we wrap the error with an internal error code? Like below (`status.Error(codes.internal, err.Error()`)